src/Kunstmaan/GeneratorBundle/Generator/PagePartGenerator.php

```diff
@@ -144,18 +149,40 @@ private function generateResourceTemplate()
     private function generateSectionConfig()
     {
         if (count($this->sections) > 0) {
-            $dir = $this->bundle->getPath() . '/Resources/config/pageparts/';
+            if ($this->isSymfony4()) {
+                $dir = $this->bundle->getPath().'/../config/kunstmaancms/pageparts/';
+            } else {
+                $dir = $this->bundle->getPath().'/Resources/config/pageparts/';
+            }
+
             foreach ($this->sections as $section) {
                 $data = Yaml::parse(file_get_contents($dir . $section));
+
+                // grab the sf4 style data if needed
+                if (array_key_exists('kunstmaan_page_part', $data) && array_key_exists('pageparts', $data['kunstmaan_page_part'])) {
+                    $pagePartKey = array_keys($data['kunstmaan_page_part']['pageparts'])[0];
+                    $data = $data['kunstmaan_page_part']['pageparts'][$pagePartKey];
+                }
+
                 if (!array_key_exists('types', $data)) {
                     $data['types'] = array();
                 }
+
                 $data['types'][] = array(
                     'name' => str_replace('PagePart', '', $this->entity),
                     'class' => $this->bundle->getNamespace() . '\\Entity\\PageParts\\' . $this->entity,
                 );
 
-                $ymlData = Yaml::dump($data);
+                // restore the sf4 style data
+                if (isset($pagePartKey)) {
+                    $data = [
+                        'kunstmaan_page_part' => [
```

Comment:
Shouldn't this only be done when in a symfony 4 app?

Reply:
$pagePartKey only exists when the symfony 4 config is used.